tasks/ruby.rb

```diff
@@ -1,46 +1,33 @@
 #!/opt/puppetlabs/puppet/bin/ruby
 # frozen_string_literal: true
 require 'open3'
+require_relative "../../ruby_task_helper/files/task_helper.rb"
 
-def facter_executable
-  if Gem.win_platform?
-    require 'win32/registry'
-    installed_dir =
-      begin
-        Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Puppet Labs\Puppet') do |reg|
-          # rubocop:disable Style/RescueModifier
-          # Rescue missing key
-          dir = reg['RememberedInstallDir64'] rescue ''
-          # Both keys may exist, make sure the dir exists
-          break dir if File.exist?(dir)
+class Facts < TaskHelper
+  def facter_executable
+    install_path = File.join(File.dirname(RbConfig.ruby), 'facter')
 
-          # Rescue missing key
-          reg['RememberedInstallDir'] rescue ''
-          # rubocop:enable Style/RescueModifier
-        end
-      rescue Win32::Registry::Error
-        # Rescue missing registry path
-        ''
-      end
-
-    facter =
-      if installed_dir.empty?
-        ''
-      else
-        File.join(installed_dir, 'bin', 'facter')
-      end
-  else
-    facter = '/opt/puppetlabs/puppet/bin/facter'
+    # Fall back to PATH lookup if puppet-agent isn't installed
+    if File.exist?(install_path)
+      # Paths with spaces must be quoted on Windows, which means slashes need escaping
+      Gem.win_platform? ? "\"#{install_path.tr('/', '\\\\')}\"" : install_path
```

Comment:
If we continue invoking facter with the `exec(*args)` form instead of with a single string, we don't need this escaping.